src/WebJobs.Script/Workers/Http/HttpWorkerChannel.cs

```diff
@@ -40,10 +43,39 @@ public Task InvokeFunction(ScriptInvocationContext context)
             return _httpWorkerService.InvokeAsync(context);
         }
 
+        internal async Task DelayUntilWokerInitialized()
+        {
+            _workerChannelLogger.LogDebug("Initializing HttpWorker.");
+            try
+            {
+                bool isWorkerReady = await _httpWorkerService.IsWorkerReady();
+                if (!isWorkerReady)
+                {
+                    PublishWorkerErrorEvent(new TimeoutException("Initializing HttpWorker timedout."));
+                }
+                _workerChannelLogger.LogDebug("HttpWorker is Initialized.");
```

Comment:
Do we still want to log this if initialization fails? Might be a bit confusing

Reply:
fixed!